common/src/main/java/com/wynntils/models/players/GuildModel.java

```diff
@@ -274,6 +278,7 @@ public void parseGuildInfoFromGuildMenu(ItemStack guildInfoItem) {
             Matcher guildNameMatcher = line.getMatcher(GUILD_NAME_MATCHER);
             if (guildNameMatcher.matches()) {
                 guildName = guildNameMatcher.group("name");
+                Models.Territory.scheduleTerritoryUpdate(true);
```

Comment:
Did originally have this as a `GuildEvent.Set` but not sure how often that would be used so changed it to this. Can change it back if that's a better approach

Reply:
What if you started scheduling with the small timer, and on every reschedule, you check the guild model, so you know whether you are in a guild or not. And then, you only need `GuildEvent.Joined` to "cancel" the current timer (and optionally `GuildEvent.Left`, but that's only a micro-optimization).

Reply:
Do you still need the whole scheduling method still? I think you really could just do one of `schedule`-ing with the `schedule` method, and have the download itself handle schedule itself after finishing.

_That's a lot of scheduling :)_